### ISSUE TITLE
Breaks the code if extend a pseudo-class starting from a colon

### ISSUE THREAD

**Issue by viktor-yakubiv** [prettier/prettier#7040]
Using an [`@extend`](https://jonneal.dev/specs/css-extend-rule/) with a pseudo-class, such as `:root`, `:focus` or any other is not correctly understood by Prettier. I assume that the tool mislead line starting with `@` and containing a colon (`:`) to a property rather than at-rule and tries to fix a colon placement. It breaks the code as you might see in the following example. 

**Prettier 1.19.1**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuZAnCEYAJgF8A6UhAFjALYA2OhWWAAnAB7xQAmWS6mA3IbiABoQEAA4wAltADOyUAENU6AO4AFBQhko5ANwjjWgkACNUcsAGs4MAMoiz4qAHNkMVAFc4Q0pQDqxcfBSdmBw1hoB4toBAJ7I4FIyQg5ScKgwKqaOZHLIAGZyFClCAFZSjABCphZW1nJkcAAyDnB5BUUgpYzWDo4UcACKbpgtSPmFniB2qCmocWAJhiKoDjA++jDEyAAcAAxCSxApPqYicUtwM9otQgCOQ-AZopogclIAtFBwcKzfhqhwd3E-wyciyOVGbQmKTI4hc7ihPT6g2GrXGQhgciMa1YG2QACZ0aZxBQegBhCBkbJxKDQa4gNwpAAqmM0Y3a2g8AEk2AgbGBlmIAIJsawwaJ9VEpXC4IA)
```sh
--parser css
```

**Input:**
```css
:root {}

html {
  @extend :root;
}
```

**Output:**
```css
:root {
}

html {
  @extend: root;
}

```

**Expected behavior:**

Block `html` does not change.

**Comment by viktor-yakubiv** [comment]
I am not completely sure is this bug of Prettier but [`postcss-extend-rule`](https://github.com/csstools/postcss-extend-rule) works fine. Please, let me know if you should contact PostCSS about.

**Comment by thorn0** [comment]
This is not a standard, right? So supporting this should be considered a feature, not a bug.

**Comment by viktor-yakubiv** [comment]
Yes, in [CSS spec](https://jonneal.dev/specs/css-extend-rule/) is only a _A Collection of Interesting Ideas_. Although, the feature is in [SCSS docs](https://sass-lang.com/documentation/at-rules/extend) already.

I [tried it out](https://prettier.io/playground/#N4Igxg9gdgLgprEAuZAnCEYAJgF8A6UhAFjALYA2OhWWAAnAB7xQAmWS6mhuIANCAgAHGAEtoAZ2SgAhqnQB3AApyEUlDIBuEUa34gARqhlgA1nBgBlISdFQA5shioArnAGlKAdWKj4EmzA4SzU-UU0-AE9kcAkpATsJOFQYJWN7MhlkADMZCiSBACsJRgAhYzMLSxkyOAAZOzgcvIKQYsZLO3sKOABFF0wmpFz89xAbVCTUGIkwOP0hVDsYL10YYmQADgAGAUWIJK9jIRjFuCnNJoEARwH4NOF1EBkJAFooODhWL-1UOFvRH80jIMllhi0xkkyKInK5IV0ev1Bs1RgIYDIDKtWOtkAAmNHGUQULoAYQgZEyMSg0CuIBcSQAKhj1CNWpo3ABJNgIKxgJYiACCbEsMEiPRRSVwuCAA) and the example above works just fine with Prettier. I see from the [docs](https://prettier.io/docs/en/configuration.html#setting-the-parserdocsenoptionshtmlparser-option) there is a way to explicitly set parser for a filetype, so passing `parser: scss` would be a workaround.

I am not closing the issue because it's up to collaborators to decide does this feature request fit your priorities. Thank you a lot!

**Comment by viktor-yakubiv** [comment]
Please, check the following example. If there is only one at-rule PostCSS considers it as a property, i.e. `css-decl`. However, if pass there another nested at-rule as in the example below, it considers `@extend` being `css-atrule`.

**Prettier 1.19.1**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuZAnCEYAJgF8A6UhAFjALYA2OhWWAAnAB7xQAmWS6mA3DfWXFYBLAIZYAFGSFQAtAHchrGMSRYAbAAYNAB0YBKHASi4QAGhARtMIdADOyUCNTo5ABScJ7KEQDcIisxAAI1QRMABrOBgAZW0w6QBzZBhUAFc4c1JKAHViIXhbOLA4aM98oR98gE9kcFt7c2lbOFQYV1CEshFkADMRCmbzACtbRgAhUIio6JEBABlpOF7+wZARxmjEijgARVTMJaQ+gYyQONRm1FqwesDtVGkYbMVlZAAODXN7iGbs0O1avc4JcfEtzABHfbwdqWLwgES2GRQOCCQSBVBwSFCDHtESdbpHFanZpSZJpYlbXZQw7HVYwERBZ5KYjIABM5hSIiEFESAGEIGQurUoNAwSBUs0ACoMry004+dIASTYCBiYAeVgAgmxojAqttlidcLggA)
```sh
--parser css
```

**Input:**
```css
:root {}

html {
  @extend :root;
  @media (min-width: 600px) {}
}
```

**Output:**
```css
:root {
}

html {
  @extend :root;
  @media (min-width: 600px) {
  }
}

```

**Expected behavior:**

Behaviour is correct.

---

 Although, if `@media` rule is different, e.g. `@media screen {}`, the result is the same as before. I consider this as a bug on PostCSS side, perhaps with parser or with `postcss-preset-env` in detecting environment.
